examples/training/classification/densenet/imagenet/basic_training.py

```diff
@@ -0,0 +1,224 @@
+    label_key = "image/class/label"
+    keys_to_features = {
+        image_key: tf.io.FixedLenFeature((), tf.string, ""),
+        label_key: tf.io.FixedLenFeature([], tf.int64, -1),
+    }
+    parsed = tf.io.parse_single_example(example, keys_to_features)
+
+    # Decode and resize image
+    image_bytes = tf.reshape(parsed[image_key], shape=[])
+    image = tf.io.decode_jpeg(image_bytes, channels=3)
+    image = tf.image.resize(image, IMAGE_SIZE)
+
+    # Decode label
+    label = tf.cast(tf.reshape(parsed[label_key], shape=()), dtype=tf.int32) - 1
+    label = tf.one_hot(label, NUM_CLASSES)
+    return image, label
+
+
+def load_imagenet_dataset():
+    train_filenames = [
+        f"{FLAGS.imagenet_path}/train-{i:05d}-of-01024" for i in range(0, 1024)
+    ]
+    validation_filenames = [
+        f"{FLAGS.imagenet_path}/validation-{i:05d}-of-00128" for i in range(0, 128)
+    ]
+
+    train_dataset = tf.data.TFRecordDataset(filenames=train_filenames)
+    validation_dataset = tf.data.TFRecordDataset(filenames=validation_filenames)
+
+    train_dataset = train_dataset.map(
+        lambda x: parse_imagenet_example(x, IMAGE_SIZE),
+        num_parallel_calls=tf.data.AUTOTUNE,
+    )
+    validation_dataset = validation_dataset.map(
+        lambda x: parse_imagenet_example(x, IMAGE_SIZE),
+        num_parallel_calls=tf.data.AUTOTUNE,
+    )
+
+    return train_dataset.batch(BATCH_SIZE), validation_dataset.batch(BATCH_SIZE)
+
+
+train, test = load_imagenet_dataset()
+
+
+"""
+Next, we augment our dataset. We define a set of augmentation layers and then apply them to our input dataset using the `apply_augmentation` method from our KerasCV training utils.
+"""
+
+AUGMENT_LAYERS = [
+    keras_cv.layers.RandomFlip(),
+    keras_cv.layers.RandAugment(value_range=(0, 255), magnitude=0.3),
+    keras_cv.layers.RandomCutout(height_factor=0.1, width_factor=0.1),
+]
+
+
+@tf.function
+def augment(img, label):
+    inputs = {"images": img, "labels": label}
+    for layer in AUGMENT_LAYERS:
+        inputs = layer(inputs)
+    return inputs["images"], inputs["labels"]
+
+
+train = train.map(augment, num_parallel_calls=tf.data.AUTOTUNE).prefetch(
+    tf.data.AUTOTUNE
+)
+
+
+"""
+Now we can begin training our model. We begin by loading a DenseNet model from KerasCV.
+"""
+
+
+def get_model():
+    return DenseNet121(
+        include_rescaling=True,
+        include_top=True,
+        num_classes=NUM_CLASSES,
+        input_shape=IMAGE_SIZE + (3,),
+    )
+
+
+"""
+Next, we pick an optimizer. Here we use Adam with a linearly decaying learning rate.
+"""
+
+
+def get_optimizer():
+    return Adam(
+        learning_rate=PolynomialDecay(
+            initial_learning_rate=0.005,
+            decay_steps=train.cardinality().numpy() * EPOCHS,
+            end_learning_rate=0.0001,
+        )
+    )
+
+
+"""
+Next, we pick a loss function. Here we use a built-in Keras loss function, so we simply specify it as a string.
+"""
+
+
+def get_loss_fn():
```

Comment:
introduce label_smoothing - which will require you to call `keras.losses.CategoricalCrossentropy(label_smoothing=0.1)`.

label_smoothing improves performance when used in conjunction with cutmix